src/total.jl

```diff
@@ -44,30 +40,60 @@ function total(x::Symbol, design::SimpleRandomSample)
     end
     m = mean(x,design)
     total = design.popsize * m.mean[1]
-    return DataFrame(total=total, se_total=se_tot(x, design))
+    return DataFrame(total=total, SE=se(x, design))
 end
 
 """
-Inner methods for `by`
-"""
-function se_total_by(x::AbstractVector, design::SimpleRandomSample, _)
-    # vector of length equal to `sampsize` containing `x` and zeros
-    z = cat(zeros(design.sampsize - length(x)), x; dims=1)
-    # variance of the total
-    variance = design.popsize^2 / design.sampsize * design.fpc * var(z)
-    # return the standard error
-    return sqrt(variance)
-end
+Estimate subpopulation total for a stratified sample. 
+```jldoctest
+julia> using  Survey;
+
+julia> apisrs = load_data("apisrs");
 
-function total(x::AbstractVector, design::SimpleRandomSample, weights)
-    total = wsum(x, weights)
-    return DataFrame(total=total, sem=se_total_by(x, design::SimpleRandomSample, weights))
+julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
+
+julia> total(:api00, :cname, srs) |> first
+DataFrameRow
+ Row │ cname     total      SE     
+     │ String15  Float64    Float64 
+─────┼──────────────────────────────
+   1 │ Kern      1.77644e5  55600.8
+
+```
+"""
+function total(x::Symbol, by::Symbol, design::SimpleRandomSample)
+    function domain_total(x::AbstractVector, design::SimpleRandomSample, weights)
+        function se(x::AbstractVector, design::SimpleRandomSample, _)
+            # vector of length equal to `sampsize` containing `x` and zeros
+            z = cat(zeros(design.sampsize - length(x)), x; dims=1)
+            variance = design.popsize^2 / design.sampsize * design.fpc * var(z)
+            return sqrt(variance)
+        end
+        total = wsum(x, weights)
+        return DataFrame(total=total, SE=se(x, design::SimpleRandomSample, weights))
+    end
+    gdf = groupby(design.data, by)
+    combine(gdf, [x, :weights] => ((a, b) -> domain_total(a, design, b)) => AsTable)
 end
 
 """
```

Comment:
This docstring should be deleted and the example should be added inside the docstring starting at line 3. These methods should be grouped together.

Reply:
Can you give committable suggestion for this?

Reply:
Or do it in the next PR.

Reply:
Not really, cause I would need to commit each row. I don't think I can make commit suggestions for multiple rows, or at least I couldn't find a way. I will commit to your fork, as you suggested. You agree with removing some docstrings and regrouping methods as I suggested?

Reply:
Let's see it in a PR.